### PR TITLE
[Performance][SpecDecode] Add FULL ACLGraph for DeepSeek V4 DSpark drafter

### PR DESCRIPTION
**Pull request by QwertyJack**
### What this PR does / why we need it?

This is PR 4/4 of the DeepSeek V4 DSpark split defined in #11126. It depends on #12005 and adds graph optimization only after the eager path is independently established.

For stacked review, the owned change is commit `386530d12` (`qj/dspark-03-eager..qj/dspark-04-aclgraph`). The temporary `main`-based GitHub diff also contains PRs 1-3 until their dependencies merge.

This PR:

- derives drafter capture sizes from complete target request rows;
- uses persistent graph input, attention metadata, sampling, and output buffers;
- pads complete inactive request rows across DP ranks;
- pads missing speculative-width metadata for graph-added request rows before masking sampled tokens;
- disables unsafe PIECEWISE replay during uneven DeepSeek V4 DSpark DP turnover while preserving FULL mode when all ranks agree;
- captures/replays the fixed-shape draft forward and slices outputs back to actual requests;
- keeps `speculative_config.enforce_eager=true` as the eager isolation path.

Incremental size: about `+1104/-35`, including 643 lines of focused graph/DP regression tests.

### Does this PR introduce _any_ user-facing change?

Yes. When the target model uses FULL ACLGraph, the DeepSeek V4 DSpark drafter uses FULL capture/replay by default. Users can set `speculative_config.enforce_eager=true` to isolate or disable the drafter graph.

### How was this patch tested?

PR 4 graph/runner-focused result: `36 passed`.

Cumulative PR 1-4 unit result:

```bash
python -m pytest -q \
  tests/ut/quantization/test_modelslim_config.py \
  tests/ut/sample/a2/test_probabilistic_rejection.py \
  tests/ut/sample/test_gumbel_noise.py \
  tests/ut/sample/test_rejection_sampler.py \
  tests/ut/spec_decode/test_deepseek_v4_draft_hooks.py \
  tests/ut/spec_decode/test_deepseek_v4_dspark_model.py \
  tests/ut/spec_decode/test_dspark_dsa_metadata.py \
  tests/ut/spec_decode/test_dspark_eager_proposer.py \
  tests/ut/spec_decode/test_dspark_aclgraph.py \
  tests/ut/spec_decode/test_llm_base_proposer.py \
  tests/ut/worker/a2/test_model_runner_v1.py
```

Result: `166 passed, 1 skipped`.

Also passed:

- mypy for all 32 Python files changed by the four-PR stack;
- `pre-commit run --all-files`;
- `ruff check` and `ruff format --check` for all changed Python files;
- `python -m py_compile` for the proposer and runner paths;
- `git diff --check qj/dspark-03-eager...HEAD`.

The final real W4A8 TP4/DP4/EP16 service, FULL replay, SPEED-Bench, coding80 AR/AL, and fixed 8k/1k performance evidence will be added before this draft is marked ready.

References:

- Direct dependency: #12005
- RFC and four-PR plan: #11126
- Original end-to-end prototype: #11196
- Generic Qwen/GLM DSpark foundation: #11765

- vLLM version: v0.23.0
- vLLM main: https://github.com/vllm-project/vllm/commit/e5588e49bc2642670116664a7fc4096e27adb179

### CODE REVIEW

File: vllm_ascend/quantization/modelslim_config.py [hunk 1]
```diff
@@ -670,7 +706,11 @@ def get_quant_method(self, layer: torch.nn.Module, prefix: str, tid2eid=None) ->
             prefix = prefix.replace("self_attn", "attention")
         if model_type in packed_modules_model_mapping:
             self.packed_modules_mapping = packed_modules_model_mapping[model_type]
-        prefix = self.quant_prefix_mapper(model_type, prefix)
+        prefix = self.quant_prefix_mapper(
+            model_type,
+            prefix,
+            num_hidden_layers=getattr(vllm_config.model_config.hf_config, "num_hidden_layers", None),
+        )
```
[the remark this answers is not in the document]
Comment (QwertyJack) [reply vllm_ascend/quantization/modelslim_config.py:713]: Not applying this suggestion because the reported NameError is not present. get_quant_method defines vllm_config = get_current_vllm_config() at line 689 in the same method, before this use. The current local variable is intentional and covered by the ModelSlim unit suite.

File: vllm_ascend/worker/model_runner_v1.py [hunk 2]
```diff
@@ -1763,6 +1841,30 @@ def _copy_valid_sampled_token_count(
...
+        spec_decode_metadata: SpecDecodeMetadata | None,
+    ) -> torch.Tensor:
+        if spec_decode_metadata is None:
+            return sampled_token_ids
+        cu_num_draft_tokens = spec_decode_metadata.cu_num_draft_tokens
+        num_draft_tokens = torch.empty_like(cu_num_draft_tokens)
+        num_draft_tokens[:1] = cu_num_draft_tokens[:1]
+        num_draft_tokens[1:] = cu_num_draft_tokens[1:] - cu_num_draft_tokens[:-1]
+        max_valid_width = num_draft_tokens[: sampled_token_ids.shape[0]] + 1
+        columns = torch.arange(
```
[the remark this answers is not in the document]
Comment (QwertyJack) [reply vllm_ascend/worker/model_runner_v1.py:1862]: Applied in 0e8469f93. Missing num_draft_tokens rows are padded with zeros before the valid-width mask is built, so graph-added rows keep only their first sampled token and mask later columns. The regression is covered by test_mask_sampled_token_ids_respects_scheduled_draft_width; the cumulative focused suite passes with 166 passed and 1 skipped.
